Small improvments / bug fixes

casperklein opened the pull request:
I hope that it's okay to wrap these little changes in one PR.

**New:**

- Allow the usage of `docker-compose build`

**Bugfixes:**

- Start DB container first and wait until it's ready, before starting phpservermonitor. I noticed before, that there is a DB connection error shown in phpservermonitor, until MariaDB is ready.
- When doing `docker-compose down / up`, the MariaDB container is deleted and the configured data was lost.

**Improvement:**

- .dockerignore added

casperklein committed:
version updated

casperklein committed:
enable 'docker-compose build'

casperklein committed:
start db container first

casperklein committed:
Use bind mount, to preserve database files on container recreation.

casperklein committed:
wait until mariadb is ready

casperklein committed:
Add only the files needed to docker build context

casperklein commented:
Maybe related: https://github.com/Quentinvarquet/docker-phpservermonitor/issues/8

casperklein commented:
@TimZ99, can you take a look?

TimZ99 commented:
@casperklein I can't get it to work. It shows `Waiting until MariaDB has started.....` until I exit by hand.

casperklein commented:
How long did you wait? It can take some time until MariaDB is started. Up to 1minute in my setup.

One other thing you could test is, change in `maria-wait.sh`

`until (timeout 1 bash -c "< /dev/tcp/database/3306") &> /dev/null; do` to something with a higher timeout:
`until (timeout 3 bash -c "< /dev/tcp/database/3306") &> /dev/null; do`

TimZ99 commented:
I waited like 10 minutes. My db shows it is ready for connections for the last 2 hours. I'm running docker on macOS 11 btw, maybe that is the reason

casperklein commented:
> I waited like 10 minutes.

That's weird. Unfortunately, I don't have a mac to test with. I tested again on a clean Debian host and it works like a charm:

```bash
git clone https://github.com/casperklein/docker-phpservermonitor
vi docker-compose.yml # set MYSQL_PASSWORD and MYSQL_ROOT_PASSWORD
docker-compose up
```

Result:

```
Building phpservermonitor
Sending build context to Docker daemon  9.728kB
Step 1/31 : FROM php:7.4-apache
[..]
phpservermonitor_mariadb | 2021-04-05 12:50:46 0 [Note] Added new Master_info '' to hash table
phpservermonitor_mariadb | 2021-04-05 12:50:46 0 [Note] mysqld: ready for connections.
phpservermonitor_mariadb | Version: '10.5.9-MariaDB-1:10.5.9+maria~focal'  socket: '/run/mysqld/mysqld.sock'  port: 3306  mariadb.org binary distribution
phpservermonitor_mariadb | 2021-04-05 12:50:46 3 [Warning] Aborted connection 3 to db: 'unconnected' user: 'unauthenticated' host: '172.19.0.3' (This connection closed normally without authentication)
phpservermonitor    | Waiting until MariaDB has started.......
phpservermonitor    | AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.19.0.3. Set the 'ServerName' directive globally to suppress this message
phpservermonitor    | AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.19.0.3. Set the 'ServerName' directive globally to suppress this message
phpservermonitor    | [Mon Apr 05 12:50:47.069373 2021] [mpm_prefork:notice] [pid 42] AH00163: Apache/2.4.38 (Debian) PHP/7.4.16 configured -- resuming normal operations
phpservermonitor    | [Mon Apr 05 12:50:47.069470 2021] [core:notice] [pid 42] AH00094: Command line: 'apache2 -D FOREGROUND'
```

> My db shows it is ready for connections for the last 2 hours

Maybe I misunderstood something. You wrote you canceled after like 10min. So how can the DB running for 2 hours? Do you use a separate DB (not the one included in the `docker-compose.yml`?).

You could try this:

`docker exec phpservermonitor bash -c 'timeout 1 bash -c "< /dev/tcp/database/3306"' && echo "OK" || echo "NOK"`

to see if there is any error. Maybe _database_ cannot be resolved?

Knowing your `docker-compose.yml` would also help.

However, if that's to much hassle, I can remove the connectivity check completely, if there rest of this PR is fine and gets merged. Just let me know.

TimZ99 committed:
Added an extra waiting message

In the Docker Desktop, the wait message will only show after its done waiting. This message wil appear first.

TimZ99 committed:
Replaced links for network

TimZ99 commented:
@casperklein Can you check if everything still works for you?

casperklein committed:
PMA env var fixed

casperklein committed:
db host fixed

TimZ99 commented:
👍🏼 great job, will merge.

casperklein commented:
@TimZ99 It's working for me 👍
